pandas/core/computation/ops.py

```diff
@@ -102,10 +103,9 @@ def evaluate(self, *args, **kwargs) -> Term:
     def _resolve_name(self):
         local_name = str(self.local_name)
         is_local = self.is_local
-        if local_name in self.env.scope and isinstance(
-            self.env.scope[local_name], type
-        ):
-            is_local = False
+        check = self.env.scope.get("column") == local_name
+        if not isinstance(check, Series) and check:
```

Comment:
Shouldn't `self.env.scope.get("column") == local_name` always produce a bool?

Reply:
yes, ideally it should. 
Although the tests were fine, but it threw the error you mentioned up here https://github.com/pandas-dev/pandas/pull/49248#issuecomment-1289588600

Reply:
Not too familiar with this part of the codebase, but is there a way to check if it's local without checking if it's a series?

Reply:
the error on the docs were specific to Series (either `local_name` or `self.env.scope.get("column")`)
One example below
![Screenshot 2022-10-31 at 11 08 42 PM](https://user-images.githubusercontent.com/6472290/199073476-4814d4ce-8885-4fd0-92a1-52db2eb8c232.png)

Reply:
Can the `self.env.scope.get("column")` Series `name` just be compared to the `local_name`?

Reply:
so something like

- check if Series
- compare self.env.scope["column"].name with local_name
- if not Series compare directly self.env.scope["column"]

sorry but how is that any better?

Reply:
(This may be my ignorance to this part of the code base), but I assumed that `self.env.scope["column"]` would refer to a Series object and the `local_name` would refer to the name of the Series so no `Series` checking needs to be done.

Reply:
Oh, no that's not the case. Usually it's name of a df column. 

I did the Series check just to cover the Sphinx errors. Maybe someone with better Sphinx knowledge can fix this more elegantly.